presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -1015,6 +1022,9 @@ else if (sparkException.getMessage().contains("Executor heartbeat timed out") ||
                                 sparkException.getMessage().contains("Unable to talk to the executor")) {
                             wrappedPrestoException = new PrestoException(SPARK_EXECUTOR_LOST, executionException);
                         }
+                        else if (errorClassifier.isPresent()) {
+                            wrappedPrestoException = errorClassifier.get().classify(executionException);
+                        }
```

Comment:
I can see the existing "hardcoded" error classification is preferred over the external `ErrorClassifier` (this is applied only when the default classifier doesn't match). It might be useful to be able to completely override the decision making here by injecting custom classifier.

Reply:
The pluggable ErrorClassifier is only intended to work on exceptions/failures that are company specific and does not apply to Open source world. If we can classify an error in a general manner (like the one happening in the code above), then we will ideally prefer that.  

So, basically, I imagine error classification to work in this way/order:
1. General Classifier: This classifier classifier any failures which are generic in nature (coming from OSS spark code) and this classifier sits in Presto OSS code
2. Custom Classifier : This classifier should classifier errors which cannot be classified by general classifier since these are specific to company's deployment or being thrown by internal services. 
3. If none of these can classify the error, throw GENERIC_SPARK_ERROR

In its current form, you can inject only 1 instance of ErrorClassifier. So, general classifier logic will have to be kind of "hardcoded" while the custom classifier can be injected via Guice bindings.

 Ideally, it would be cleaner to move this whole error handling logic in a separate file where we can follow the above order of classifying the error. If you prefer that, I can do it in a subsequent diff.

Reply:
Definitely not a blocker. Just a suggestion. I'll leave this decision up to the Presto maintainers.

Reply:
I think extracting the error handling logic is a nice follow up.  I feel okay merging this for now first.